mirai-core-mock/src/internal/remotefile/v2/MockAbsoluteFile.kt

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2019-2022 Mamoe Technologies and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license that can be found through the following link.
+ *
+ * https://github.com/mamoe/mirai/blob/dev/LICENSE
+ */
+@file:Suppress("invisible_member", "INVISIBLE_REFERENCE")
+
+package net.mamoe.mirai.mock.internal.remotefile.v2
+
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.firstOrNull
+import net.mamoe.mirai.contact.FileSupported
+import net.mamoe.mirai.contact.file.AbsoluteFile
+import net.mamoe.mirai.contact.file.AbsoluteFolder
+import net.mamoe.mirai.internal.message.FileMessageImpl
+import net.mamoe.mirai.message.data.FileMessage
+import net.mamoe.mirai.mock.utils.mock
+
+internal class MockAbsoluteFile(
+    override val sha1: ByteArray,
+    override val md5: ByteArray,
+    private val files: MockRemoteFiles,
+    override var parent: AbsoluteFolder?,
+    override val id: String,
+    override var name: String,
+    override var absolutePath: String,
+    override val contact: FileSupported = files.contact,
+    override var expiryTime: Long = 0L,
+    override val size: Long = 0,
+    override val isFile: Boolean = true,
+    override val isFolder: Boolean = false,
+    override val uploadTime: Long = 0,
+    override var lastModifiedTime: Long = 0,
+    override val uploaderId: Long = 0
+) : AbsoluteFile {
+    private var _exists = true
```

Comment:
```suggestion
    @Volatile
    private var _exists = true
```

mirai 支持多线程工作, 尽量避免出现 data race 问题